RecoTauTag/HLTProducers/src/L2TauTagNNProducer.cc

```diff
@@ -732,7 +732,10 @@ void L2TauNNProducer::fillPatatracks(tensorflow::Tensor& cellGridMatrix,
 
 std::vector<float> L2TauNNProducer::getTauScore(const tensorflow::Tensor& cellGridMatrix) {
   std::vector<tensorflow::Tensor> pred_tensor;
-  tensorflow::run(L2cacheData_->session, {{inputTensorName_, cellGridMatrix}}, {outputTensorName_}, &pred_tensor);
+  // Check for empty input
+  if (cellGridMatrix.NumElements() != 0) {
```

Comment:
It is guaranteed because `nTaus = cellGridMatrix.shape().dim_size(0)`, which is the same check, but I agree that it can be improved by wrapping all the snippet around a `if (nTaus > 0)`. 
If you wish I can rework it quickly

Reply:
>  If you wish I can rework it quickly

perhaps leave a comment, if it's guaranteed let's not add more if branches.
Thank you.